src/Orleans.Transactions/DistributedTM/ParticipantId.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Orleans.Concurrency;
+using Orleans.Runtime;
+using Orleans.Transactions.Abstractions;
+
+namespace Orleans.Transactions
+{
+    [Serializable]
+    [Immutable]
+    public readonly struct ParticipantId
+    {
+        public static readonly IEqualityComparer<ParticipantId> Comparer = new IdComparer();
+
+        [Flags]
+        public enum Role
+        {
+            Resource = 1 << 0,
+            Manager = 1 << 1,
+            PriorityManager = 1 << 2
+        }
+
+        public string Name { get; }
+        public GrainReference Reference { get; }
+        public Role SupportedRoles { get; }
```

Comment:
It may be nice for application to be able to specify priority TM even if not using TOC. I certainly have encountered this need in my applications (e.g. for my TPC-C workload I wanted to control which grain is the TM).

Reply:
I would keep it simple and just add a configuration option "PriorityTM" to the configuration object for the TransactionalState facet.